spec/plugins/data_table/server_side.js

```diff
@@ -433,4 +434,57 @@ describe("DataTable Plugin", function() {
     });
   });
 
+  describe("filtering", function() {
+    beforeEach(function() {
+      table = new app.Views.T({ collection : collection });
+      table.render();
+      jasmine.Ajax.requests.mostRecent().response(mockResponse.get());
+    });
+
+    it("should have filtering info for filterable column in column manager", function() {
+      var cg = table._columnManager._configGenerator;
+      var numCols = 0;
+      var numFilters = 0;
+      for (var i = 0; i < cg.columnsConfig.length; i++) {
+        var col = cg.columnsConfig[i];
+        //console.log(JSON.stringify(col));
+        expect(col).not.toEqual(null);
+        if (col) {
+          numCols++;
+          if (col.filter)
+            numFilters++;
+        }
+      }
+      //console.log("Number of columns found in config: "+numCols);
+      //console.log("Number of columns with filters:  "+numFilters);
+      expect(numCols).toEqual(4);
+      expect(numFilters).toEqual(2);
+    });
+
+    it("shouldn't create filter inputs for unfilterable columns", function() {
+      var hasFilter = [];
+      table.dataTable.find("thead th").each(function (index) {
+        hasFilter.push(this.getElementsByClassName('DataTables_filter_wrapper').length > 0);
+      });
+      expect(hasFilter).toEqual([false, true, true, false]);
+    });
+
+    it("should track filtering in column manager and in ext_filter_json parameter", function() {
+      var cg = table.configGenerator();
+      //var cg = table._columnManager.columnsConfig();
+      table.dataTable.find("thead th").each(function (index) {
+        var title = this.outerText;
+        title = title.substr(0, title.indexOf('Filter'));
+        var col = cg.columnConfigByTitle.attributes[title];
+        if (col && col.filter) {
+          if (col.filter.type == "string") {
+            $('#menu-name input', this).val("Scott").trigger("change");
+            expect(col.filter.value).toEqual("Scott");
+            expect(jasmine.Ajax.requests.mostRecent().url).toMatch("ext_filter_json");
```

Comment:
Cool. We would probably want verify the values of the filter, not only if it exists.